module2/task4.py

```diff
@@ -0,0 +1,27 @@
+l = [1, [], 2, [-19, 700, 0, [90, 33, [18, 77, [0, ], -2], 11, 16], -100]]
+
+max_value = l[0]
+min_value = l[0]
+sum_of_elem = 0
+count_of_elem = 0
+
+
+def iterate_list(list_of_elem):
+    global max_value, min_value, sum_of_elem, count_of_elem
+    for elem in list_of_elem:
+        if isinstance(elem, list):
+            if len(elem) != 0:
+                iterate_list(elem)
+            else:
+                continue
+        else:
+            max_value = max(elem, max_value)
+            min_value = min(elem, min_value)
+            sum_of_elem += elem
+            count_of_elem += 1
+
+
+iterate_list(l)
+print(f"Max value from the list is {max_value}")
+print(f"Min value from the list is {min_value}")
+print("Average value from the list is ", sum_of_elem / count_of_elem)
```

Comment:
Looks good enough. The other approach might be to make a regular list first (convert list of lists to single list). And afterwards to call min max etc. on that list, so you wouldn't need to create 4 extra variables.